flow/layers/layer.h

```diff
@@ -44,19 +44,32 @@ enum Clip { none, hardEdge, antiAlias, antiAliasWithSaveLayer };
 
 class ContainerLayer;
 
-struct PrerollContext {
+struct RasterContext {
   RasterCache* raster_cache;
   GrContext* gr_context;
-  ExternalViewEmbedder* view_embedder;
-  MutatorsStack& mutators_stack;
   SkColorSpace* dst_color_space;
-  SkRect cull_rect;
-
-  // The following allows us to paint in the end of subtree preroll
   const Stopwatch& raster_time;
   const Stopwatch& ui_time;
   TextureRegistry& texture_registry;
   const bool checkerboard_offscreen_layers;
+};
+
+using RasterOperation = std::function<void(RasterContext* raster_context)>;
+
+class RasterOperations {
```

Comment:
maybe I should call this `RasterPrerollOperation`?

Reply:
Or `PrerollRasterOperations`? Just for considerations. I don't have a strong opinion on namings :-)